app/src/main/java/com/yacgroup/yacguide/PartnersActivity.kt

```diff
@@ -207,25 +193,52 @@ class PartnersActivity : AppCompatActivity() {
             setNegativeButton(R.string.cancel) { dialog, _ -> dialog.dismiss()}
             setPositiveButton(R.string.ok) { _, _ ->
                 _db.deletePartner(partner)
+                _selectedPartnerIds.remove(partner.id)
                 _displayContent()
             }
         }
         dialog.show()
     }
 
-    private fun _updatePartner(dialog: Dialog, newName: String, partner: Partner?) {
-        when {
-            newName == "" ->
-                Toast.makeText(dialog.context, R.string.hint_no_name, Toast.LENGTH_SHORT).show()
-            _db.getPartnerId(newName) > 0 ->
-                Toast.makeText(dialog.context, R.string.hint_name_already_used, Toast.LENGTH_SHORT).show()
-            else -> {
-                val updatedPartner = partner ?: Partner()
-                updatedPartner.name = newName
-                _db.addPartner(updatedPartner)
-                dialog.dismiss()
-                _displayContent()
+    private fun _updatePartner(partner: Partner?, dialogTitle: String) {
+        val view = layoutInflater.inflate(R.layout.add_partner_dialog, null).let { view ->
+            partner?.let {
+                view.findViewById<EditText>(R.id.addPartnerEditText).setText(it.name)
             }
+            view
+        }
+        val builder = AlertDialog.Builder(this).apply {
+            setTitle(dialogTitle)
+            setView(view)
+            setNegativeButton(R.string.cancel) { dialog, _ -> dialog.dismiss()}
+            // Set to null. We override the onclick below.
+            setPositiveButton(R.string.ok, null)
         }
+        // This solution prevents that the dialog is closed automatically,
```

Comment:
Thx for documenting this. It seems that closing the dialog after clicking on negative/positive button is default behavior? If so, why is there a `dialog.dismiss()` in `setNegativeButton`? Same for `deletePartner`dialog

Reply:
You are right, We can skip this and just set the listener to `null`.